### PR TITLE
Sequence log messages from multiple sources

### PR DESCRIPTION
**Pull request by spenced**

### CODE REVIEW

File: src/vse_sync_pp/sequence.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    The output written to stdout is multiplexed content, with lines sequenced as
+    follows. An initial set of candidate messages to write is constructed by
+    parsing the first log message from each source. The message with the lowest
+    timestamp in this set is written to stdout before being replaced by the next
+    log message from its source. This process is repeated until all sources are
+    empty.
+    """
+    aparser = ArgumentParser(description=main.__doc__)
+    aparser.add_argument(
+        '--include', choices=PARSERS, nargs='*', default=(),
```
Comment (jnunyez) [inline src/vse_sync_pp/sequence.py:150]: do we need --include having --exclude and if by default we include everything?

File: src/vse_sync_pp/sequence.py [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+        help='restrict output messages to these parsers',
+    )
+    aparser.add_argument(
+        '--exclude', choices=PARSERS, nargs='*', default=(),
```
Comment (jnunyez) [inline src/vse_sync_pp/sequence.py:154]: working good 

```
cat bar | env PYTHONPATH=src python3 -m vse_sync_pp.sequence sources.yml
{"id": "gnss/time-error", "data": [1111.222, 4, 0]}
{"id": "gnss/time-error", "data": [1112.222, 5, 0]}
{"id": "gnss/time-error", "data": [1113.222, 6, 0]}
{"id": "dpll/time-error", "data": [1234.567, 3, 3, 77.0]}
{"id": "dpll/time-error", "data": [1235.567, 4, 4, 78.0]}
{"id": "gnss/time-error", "data": [1333.444, 7, 0]}
{"id": "gnss/time-error", "data": [1334.444, 8, 0]}
{"id": "gnss/time-error", "data": [1334.444, 9, 0]}
{"id": "gnss/time-error", "data": [2345.444, 1, 0]}
{"id": "dpll/time-error", "data": [2345.671, 9, 9, 56.0]}
{"id": "gnss/time-error", "data": [2346.444, 2, 0]}
{"id": "gnss/time-error", "data": [2347.444, 3, 0]}
{"id": "gnss/time-error", "data": [2348.444, 4, 0]}
{"id": "dpll/time-error", "data": [3456.712, 7, 7, 6.0]}
{"id": "gnss/time-error", "data": [3457.444, 5, 0]}
{"id": "gnss/time-error", "data": [3458.444, 6, 0]}

```
```
cat bar | env PYTHONPATH=src python3 -m vse_sync_pp.sequence sources.yml --exclude gnss/time-error
{"id": "dpll/time-error", "data": [1234.567, 3, 3, 77.0]}
{"id": "dpll/time-error", "data": [1235.567, 4, 4, 78.0]}
{"id": "dpll/time-error", "data": [2345.671, 9, 9, 56.0]}
{"id": "dpll/time-error", "data": [3456.712, 7, 7, 6.0]}
```
returns empty string (which is what we want):
` cat bar | env PYTHONPATH=src python3 -m vse_sync_pp.sequence sources.yml --exclude gnss/time-error dpll/time-error
`

File: src/vse_sync_pp/sequence.py (hunk 1 above)
Comment (spenced) [reply src/vse_sync_pp/sequence.py:150]: Strictly speaking, no, as they're both ways of restricting. It's just a convenience in case you generate both.

**Review by crwr45 — CHANGES_REQUESTED** [review]
It should be explicit in `main.__doc__` that the sequencing is done only across multiple sources, not to lines within an individual source.

That is; the lines in any one input will have their relative order maintained in the output.

**Comment by spenced** [comment]
@crwr45 I've extended the docstring to make this explicit.

**Review by crwr45 — APPROVED** [review]

**Pull request merged**